iree/hal/local/elf/testdata/BUILD

```diff
@@ -14,7 +14,6 @@ package(
 
 c_embed_data(
     name = "simple_mul_dispatch",
-    testonly = True,
```

Comment:
you can set testonly cc_binary (TESTONLY in cmake) - knowing what targets are just for testing is useful when disabling all tests

Reply:
If I tag it as TESTONLY, the binary will be bypassed if I set `-DIREE_BUILD_TESTS=OFF`, and at this point, I need to disable `IREE_BUILD_TESTS` in bare-metal config until we can finish https://github.com/google/iree/issues/3848, otherwise https://github.com/google/iree/blob/main/iree/testing/status_matchers.h will give me all kinds of hell. :(

I will change the name to indicate the binary is for tests, but I can't tag it as testonly for the bare-metal usage.